e2e-test-server/endtoendserver/server.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2019 OpenTelemetry Authors
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endtoendserver
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"sync"
+
+	"cloud.google.com/go/pubsub"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/genproto/googleapis/rpc/code"
+
+	texporter "github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/trace"
+)
+
+// Server is an end-to-end test service.
+type Server struct {
+	pubsubClient  *pubsub.Client
+	traceProvider *sdktrace.TracerProvider
```

Comment:
There should be a whole new Provider pipeline for each scenario so they are completely isolated. Then you wait for `Shutdown()` to finish before responding to the test runner at the end of the scenario. That way the spans are written when the assertions begin

Reply:
I've changed this to 0xc0d's suggestion, so this comment is out-of-date.

Reply:
Done.